File: functionTests/diagoTests/test1.R
.T_diago <- function(v){
  if (length(v)==1){ res = v }
  else { res = diag(v)}
  res
}

a = c(1,2,3)
b = c(1)
c = c(1,2,3,4,5,6,7)
d = c(1.2, 3.4, 5.6)
e = matrix(c(1,2,3,4,5,6), 2,3, byrow=T)
f = matrix(c(1,2,3,4,5,6,7,8,9), 3, 3, byrow=TRUE)

tests = list(a = a, b=b, c=c, d=d, e=e, f=f)

for(i in tests){
  print(.T_diago(i))
}

File: functionTests/diagoTests/test1.py
import numpy as np

def _T_diago(v):
    if len(v) == 1:
        res = v
    else:
        res = np.diag(v)

    return res

a = np.array([1,2,3])
b = np.array([1])
c = np.array([1,2,3,4,5,6,7])
d = np.array([1.2, 3.4, 5.6])
e = np.array([[1,2,3], [4,5,6]])
f = np.array([[1,2,3], [4,5,6], [7,8,9]])

tests = [a,b,c,d,e,f]

for i in tests:
    print(_T_diago(i))